src/ui/components/SearchInput/index.js

```diff
@@ -0,0 +1,69 @@
+/* global document, window */
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+import './style.scss';
+
+class SearchInput extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    inputRef: PropTypes.func,
+    name: PropTypes.string.isRequired,
+    placeholder: PropTypes.string,
+  }
+
+  componentDidMount() {
+    this.setIconPosition();
+    window.addEventListener('resize', this.setIconPosition);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIconPosition);
+  }
+
+  onBlur = () => {
+    if (!this.input.value) {
+      this.root.classList.remove('SearchInput--text');
```

Comment:
I'm not sure if we're using it anywhere else but it feels wrong to call it `SearchInput-text` since there's also a `SearchInput-input`.

I know we had discussed using something like [suitcss's naming conventions](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md) but we weren't definite on the descendant and utility class name conventions. It would seem I agree with their conventions.